### PR TITLE
fix(BUX-242): fix go-paymail routes

### PR DESCRIPTION
**Pull request by vitalibalashka**
Fixes all endpoints which require some data from URL or Body

### CODE REVIEW

File: server/p2p_receive_transaction_request_parser.go [hunk 1]
```diff
@@ -27,46 +28,21 @@ func parseP2pReceiveTxRequest(c *Configuration, parms url.Values, format p2pPayl
 		incomingPaymailDomain: domain,
 	}
 
-	requestTx := paymail.P2PTransaction{}
-
-	requestTx.Reference = parms.Get("reference")
-	if len(requestTx.Reference) == 0 {
-		return nil, &parseError{ErrorMissingField, "missing parameter: reference"}
-	}
```
Comment (arkadiuszos4chain) [inline server/p2p_receive_transaction_request_parser.go:35]: Why did you remove valiadtion of required fields?

Comment (vitalibalashka) [reply server/p2p_receive_transaction_request_parser.go:35]: I have no idea. Accidentally.
Thanks for noticing. I'll return it

**Review by kuba-4chain — APPROVED** [review]

File: server/p2p_receive_transaction_request_parser.go (hunk 1 above)
Comment (vitalibalashka) [reply server/p2p_receive_transaction_request_parser.go:35]: validations are back

**Review by arkadiuszos4chain — APPROVED** [review]
